src/Native/Unix/System.Security.Cryptography.Native.Apple/pal_x509chain.cpp

```diff
@@ -0,0 +1,262 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "pal_x509chain.h"
+
+#ifndef DEBUG_CHAIN
+static int not_printf(const char* format, ...)
+{
+    (void)format;
+    return 0;
+}
+#define printf not_printf
+#endif
+
+extern "C" SecPolicyRef AppleCryptoNative_X509ChainCreateDefaultPolicy()
+{
+    return SecPolicyCreateBasicX509();
+}
+
+extern "C" SecPolicyRef AppleCryptoNative_X509ChainCreateRevocationPolicy()
+{
+    return SecPolicyCreateRevocation(kSecRevocationUseAnyAvailableMethod);
+}
+
+extern "C" int32_t
+AppleCryptoNative_X509ChainCreate(CFTypeRef certs, CFTypeRef policies, SecTrustRef* pTrustOut, int32_t* pOSStatus)
+{
+    if (pTrustOut != nullptr)
+        *pTrustOut = nullptr;
+    if (pOSStatus != nullptr)
+        *pOSStatus = noErr;
+
+    if (certs == nullptr || policies == nullptr || pTrustOut == nullptr || pOSStatus == nullptr)
+        return -1;
+
+    *pOSStatus = SecTrustCreateWithCertificates(certs, policies, pTrustOut);
+    return *pOSStatus == noErr;
+}
+
+static void DumpDict(CFStringRef formatString, ...)
```

Comment:
Use an #ifdef around this to make it a no-op when not debugging (like with printf)?